homeassistant/components/todoist/calendar.py

```diff
@@ -169,48 +199,49 @@ def setup_platform(
         # Create the custom project and add it to the devices array.
         project_devices.append(
             TodoistProjectEntity(
-                hass,
                 project,
                 labels,
                 api,
-                project_due_date,
-                project_label_filter,
-                project_id_filter,
+                utc_offset_hours=utc_offset_hours,
+                due_date_days=project_due_date,
+                whitelisted_labels=project_label_filter,
+                whitelisted_projects=project_id_filter,
             )
         )
-    add_entities(project_devices)
 
-    def handle_new_task(call: ServiceCall) -> None:
+    async_add_entities(project_devices)
+
+    async def handle_new_task(call: ServiceCall) -> None:
         """Call when a user creates a new Todoist Task from Home Assistant."""
         project_name = call.data[PROJECT_NAME]
         project_id = project_id_lookup[project_name]
 
         # Create the task
-        item = api.items.add(call.data[CONTENT], project_id=project_id)
+        content = call.data[CONTENT]
+        data = {"project_id": project_id}
 
         if LABELS in call.data:
             task_labels = call.data[LABELS]
             label_ids = [label_id_lookup[label.lower()] for label in task_labels]
-            item.update(labels=label_ids)
+            data["label_ids"] = label_ids
```

Comment:
Slight change here based on the docs: https://developer.todoist.com/rest/v1/#create-a-new-task

Reply:
Can be shortened further to:
```
        if task_labels := call.data.get(LABELS):
            data["label_ids"] = [label_id_lookup[label.lower()] for label in task_labels]
```